### PR TITLE
API Support for Legacy PDS3 DOIs

### LINKED ISSUE #180: As an API user I want to filter on PDS3 Data Set IDs with wildcards
For more information on how to populate this new feature request, see the PDS Wiki on User Story Development:

https://github.com/NASA-PDS/nasa-pds.github.io/wiki/Issue-Tracking#user-story-development

## Motivation
...so that I can search the database or a PDS3 data set by its DATA_SET_ID.

## Additional Details
This may be handled by #177 , but want to make sure we can handle searching for other identifiers besides PDS4 LIDVIDs. this may just be sticking a wildcard (\*) before the input data set ID, since in the registry we form them as `urn:nasa:pds:<data_set_id>`

## Acceptance Criteria
**Given** *<a condition>*
**When I perform** *<an action>*
**Then I expect** *<the result>*

### LINKED ISSUE #192: As a user, I want the application to support the history of PDS's DOIs, especially the one created for PDS3 products
## Motivation
...so that the application works for production.

## Additional Details

To initialize the transaction database with the history of DOI, run:

./pds_doi_service/core/util/initialize_production_deployment.py -i https://www.osti.gov/iad2/api/records -s loubrieu@jpl.nasa.gov -d venv/doi.db

Request to these DOIs do not work: 

- http://localhost:8085/PDS_APIs/pds_doi_api/0.1/dois/LRO-L-MRFLRO-2%2F3%2F5-BISTATIC-V3.0 
- http://localhost:8085/PDS_APIs/pds_doi_api/0.1/dois/LRO-L-MRFLRO-4-CDR-V1.0 return error 500

## Acceptance Criteria
**Given** the DOI service database containing all exist PDS DOIs
**When I perform** a query for a DOI with PDS3 Data Set ID as an identifier
**Then I expect** the DOI record to be returned from the database

### PR DESCRIPTION
**Pull request by collinss-jpl**
**Summary**
This branch overhauls the `initialize_production_deployment.py` script, so that it may be used to import existing DOI entries from OSTI to a local transaction database for use with the DOI service. There are also some minor fixes to handling of identifiers assigned to PDS3 label entries, such that they work correctly with the DOI Service API.

While testing import of the existing entries from OSTI, it was observed that the identifier fields for several entries contained extraneous whitespace, and that several PDS3 id's contained forward slashes (/), which interfere with the API routing performed for the GET /dois/{lidvid} endpoint. LIDVIDs (or any identifier) parsed from labels are now tidied up to remove any extra whitespace, as well as replace any forward slashes with hyphens.

**Test Data and/or Report**
There were no updates to existing regression tests for this update
[test.txt](https://github.com/NASA-PDS/pds-doi-service/files/6474643/test.txt)

After running `initialize_production_deployment.py` against the production OSTI server, the following entries identified in #192 should now be reachable:

Screen Shot 2021-05-13 at 12 25 46 PM
Note that the "LIDVID" used here is LRO-L-MRFLRO-2-3-5-BISTATIC-V3.0 (originally LRO-L-MRFLRO-2/3/5-BISTATIC-V3.0)

Screen Shot 2021-05-13 at 12 26 41 PM

Additionally, searching with wildcards via the GET /dois endpoint should also work correctly for PDS3 identifiers:
Screen Shot 2021-05-13 at 12 29 26 PM

**Related Issues**
Resolves #180 (existing wildcard support already works for PDS3 identifiers)
Fixes #192
<!--
    Reference related issues here and use `Fixes` or `Resolves` in order to automatically close the issue upon merge. For more information on autolinking to tickets see https://docs.github.com/en/github/writing-on-github/autolinked-references-and-urls.

    * for issues in this repo:
        - fixes #1
        - fixes #2
        - refs #3
    * for issues in other repos: NASA-PDS/my_repo#1, NASA-PDS/her_repo#2
->

### CODE REVIEW

File: pds_doi_service/core/outputs/osti_web_parser.py [hunk 1]
```diff
@@ -362,6 +372,16 @@ def get_lidvid_from_json(record):
                 "'report_numbers','site_url'] fields"
             )
 
+        if lidvid:
```
Comment (nutjob4life) [inline pds_doi_service/core/outputs/osti_web_parser.py:375]: If I see this block a _third_ time, I'll demand a refactoring! 😅

File: pds_doi_service/core/util/initialize_production_deployment.py [hunk 2]
```diff
@@ -1,366 +1,388 @@
...
-                        required=False)
-    parser.add_argument("-s", "--submitter_email", help="The email address of the user performing the initialization of deployment database.",
-                        required=True)
-    parser.add_argument("-d","--db_name", help="Optional database name instead of one from config.",
-                        required=False)
-    parser.add_argument("--dry-run", help="Flag to suppress the writing of DOIs to database.",
-                        required=False,action="store_true")
-    parser.add_argument("--debug", help="Flag to print debug statements.",
-                        required=False,action="store_true")
+        description='Script to import existing DOIs from OSTI into the local'
```
Comment (nutjob4life) [inline pds_doi_service/core/util/initialize_production_deployment.py:87]: Personally, I think the longer statement lines are far more readable than blind adherence to PEP8's "80 columns" rule but 🤷‍♀️

**Review by nutjob4life — APPROVED** [review]
🛳it!

**Review by tloubrieu-jpl — APPROVED** [review]
This pull request is good.

Another ticket will be created to handle pds3 identifiers with a / without replacing it.

**Pull request merged**